### PR TITLE
Optimize ht

### PR DESCRIPTION
**Pull request by Jefferyy-Peng**
- reset model after each cv fold
- add dataloader shuffle to train data loader

### CODE REVIEW

**Review by ApocalyVec — APPROVED** [review]

File: renaanalysis/utils/Training_utils.py
Comment (ApocalyVec) [inline renaanalysis/utils/Training_utils.py:1]: rename this file to training_utils following snake case naming convention

**Review by ApocalyVec — CHANGES_REQUESTED** [review]

File: renaanalysis/learning/HT.py
Comment (ApocalyVec) [inline renaanalysis/learning/HT.py:1]: Move the changes to HT to a separate self-supervised HT class.

File: renaanalysis/learning/train.py [hunk 1]
```diff
@@ -385,15 +271,21 @@ def eval(model, X, Y, criterion, last_activation, _encoder, test_name='', verbos
...
 
-    return _run_one_epoch(model, test_dataloader, criterion, last_activation, optimizer=None, mode='val', device=device, test_name=test_name, verbose=verbose)
+    test_dataloader = DataLoader(test_dataset, batch_size=batch_size)
 
+    if test_name == TestName.OddBallPreTrain.value:
+        return _run_one_epoch_self_sup(model, test_dataloader, criterion, optimizer=None, mode='val', device=device, test_name=test_name, verbose=verbose)
+    elif test_name == TestName.Normal.value or test_name == TestName.FineTune.value:
+        return _run_one_epoch_classification(model, test_dataloader, criterion, last_activation, optimizer=None, mode='val', device=device, test_name=test_name, verbose=verbose)
 
-def cv_train_test_model(X, Y, model, test_name="CNN", n_folds=10, lr=1e-4, verbose=1, l2_weight=1e-6, lr_scheduler_type='exponential', X_test=None, Y_test=None, plot_histories=False, is_test=True):
```
Comment (ApocalyVec) [inline renaanalysis/learning/train.py:396]: separate self-supervised training from cv_train_test_model, cv_train_test_model is only for training for classification task. cv_train_test_model should return test_auc, test_loss, test_acc in addition to other stuff

**Review by ApocalyVec — CHANGES_REQUESTED** [review]
